### ISSUE TITLE
-Wvla in __compiletime_assert

### ISSUE THREAD

**Issue by nickdesaulniers** [ClangBuiltLinux/linux#26]
Sedat and @kees noted that Clang is producing lots of `-Wvla` warnings now for the `__compiletime_assert` and `BUG_ON` macros.  It seems that there's some differences in Integer Constant Expression definitions between compilers, and dependence on DCE in the kernel code.

**Comment by dileks** [comment]
I haven't tried to add -Wno-vla to my compiler wrapper-script.
https://clang.llvm.org/docs/DiagnosticsReference.html#wvla

**Comment by dileks** [comment]
The most of these VLA warnings comes from the drm/i915 driver.

As a workaround I did...

[ drivers/gpu/drm/i915/Makefile ]
# clang warning
...
subdir-ccflags-y += $(call cc-disable-warning, vla)

**Comment by dileks** [comment]
i also see...
```
./include/linux/compiler.h:306:25: note: expanded from macro '__compiletime_error_fallback'
```

**Comment by nickdesaulniers** [comment]
kernel patch posted: https://lore.kernel.org/r/1534657880-11573-1-git-send-email-yamada.masahiro@socionext.com/

**Comment by nickdesaulniers** [comment]
v2: https://lore.kernel.org/r/1535220989-27645-1-git-send-email-yamada.masahiro@socionext.com/

**Comment by dileks** [comment]
I have v2 in my current patchset-4.19.

**Comment by nathanchance** [comment]
The patch that fixes this issue (same as #8) has been merged: 81b45683487a51b0f4d3b29d37f20d6d078544e4
